### DISCUSSION TITLE
Request check won't work if a Global Scope applied

### DISCUSSION THREAD

**Discussion by OptimumTasarim** [livewire/livewire#4417] [7. Bugs (archived)]
I applied a global scope to check the application's locale, but as you can guess I don't want to apply it for admin routes. Admin should see all models despite their locales.

So, here is the middleware for locale check:

```php
class LocaleScope implements Scope
{
  public function apply(Builder $builder, Model $model)
  {
    if (! request()->routeIs('admin*')) {
      $builder->where('locale', session('locale'));
    }
  }
}
```

There is no problem once the livewire component is initialized. But if I update something in the component, the if statement in the global scope `if (! request()->routeIs('admin*'))` stops working, and global scope is applied for admin routes, too.

I could resolve this by checking the user's roles or privileges but this time admin user will see whole models in the frontend, and this will screw up the frontend. For example, the admin user will see all menu items from all locales in the navbar and the navbar will blow up.

**Comment by jonneroelofs** [comment]
Have look here at : https://laravel-livewire.com/docs/2.x/security

It describes the concept of 'persistent middleware' and how you can make sure that certain middleware not only gets applied to the initial but also to subsequent requests. 

So you probably need to add you middleware in a service provider (from the docs):
```php
Livewire::addPersistentMiddleware([
    YourOwnMiddleware::class,
]);
```

**Reply by OptimumTasarim** [reply]
Thanks for the response. But my question was related to Global Scopes, not middleware.

As I stated in the first post, global scope (not middleware) is applied to the models to filter locales. But I want to exclude admin routes `/admin*` because admins should see all models. But in subsequent requests, my admin route exclusion `if (! request()->routeIs('admin*'))` won't work.

**Comment by joshhanley** [comment]
@OptimumTasarim global scopes based on the route like that won't work with Livewire, as any subsequent requests are made as an API call by Livewire to the `/livewire/message` endpoint, hence why your if statement doesn't work after the component is updated.

I'm not sure what you mean by checking users roles will cause issues as that is how I would do it personally. On the page where I need admin to see all, I'd just do something like 

```php
if ($userIsAdmin) {
    // Get all without locale
} else {
    // Get only ones with locale
}
```

Hope this helps!

**Comment by fastsol** [comment]
@OptimumTasarim did you ever figure this out?  I am running into the same issue.
